lib/trento/infrastructure/prometheus/adapter/prometheus_api.ex

```diff
@@ -7,9 +7,95 @@ defmodule Trento.Infrastructure.Prometheus.PrometheusApi do
 
   alias Trento.Repo
 
+  alias Trento.Infrastructure.Prometheus.ChartIntegration
+
   require Logger
 
   @behaviour Trento.Infrastructure.Prometheus.Gen
+  @behaviour Trento.Charts.HostDataFetcher
+
+  def ram_total(host_id, from, to) do
```

Comment:
We could sort by alphabet all the functions :) just to improve the readability slightly, fully up to you :)